### DISCUSSION TITLE
[github.dev] Is there a way to scope GitHub.dev to a subfolder?

### DISCUSSION THREAD

**Discussion by seesharprun** [community/community#77237] [Codespaces]
### Select Topic Area

Product Feedback

### Body

Today, I use GitHub.dev to make minor changes to a file in a large repository (specifically, https://github.com/microsoftdocs/azure-docs).

If I'm using GitHub Codespaces, I can usually mount a subfolder of the repository so I can "focus" my work on a single folder and avoid any issues VS Code for the Web has with large repositories.

Is there a way, with GitHub.dev, to only open a subfolder within a repository? I couldn't find any documentation on URL permutations or query string parameters to configure the folder that is mounted in the editor.

**Comment by mb21** [comment]
I'm wondering the same. In VSCode generally, and on vscode.dev, you can use workspaces for this. But seemingly not on github.dev? https://github.com/orgs/community/discussions/158557

**Comment by lanhuongsocute** [comment]
Hi, at the moment GitHub.dev doesn't support directly opening a specific subfolder within a repository through URL parameters or query strings. Unlike GitHub Codespaces, which allows you to mount and work within a subfolder to improve performance in large repositories, GitHub.dev always opens the root of the repo. Unfortunately, there’s currently no documented method to limit the mounted scope to a single subfolder in GitHub.dev. If your goal is to minimize load time or stay focused on a smaller part of a large repo, using GitHub Codespaces remains the best option. For now, you might consider submitting a feature request to GitHub Support or the Feedback form to encourage future support for subfolder targeting.

**Comment by EmirCobanOfficial** [comment]
Hi, I looked into this as well since I often work with large repositories.

As of now, GitHub.dev does not support opening a specific subfolder directly it always loads the root of the repository. I couldn’t find any URL parameters or query options for subfolder-level editing either.

Possible Workarounds:
You can manually collapse folders in the file explorer to reduce clutter and focus on the directory you need.

If performance is an issue, consider using GitHub Codespaces, which allows mounting and working inside a specific folder.

Alternatively, you can clone the repository locally and use VS Code with --folder-uri or simply open the subfolder manually.

This could definitely be a useful feature, so I’d suggest submitting product feedback or upvoting similar requests. It would help make GitHub.dev more flexible for monorepos or large documentation projects.

Hope this helps.
